domaincommon/src/state.rs

```diff
@@ -165,6 +184,38 @@ impl MatchState {
         state.turn_status.taken.insert(mech.to_string());
         state
     }
+
+    fn reduce_mech_actions(state: &MatchState, mech: &str, points_consumed: &u32) -> MatchState {
+        let mut state = state.clone();
+        let mech_state = state.mechs[mech].clone();
+        state.mechs.insert(
+            mech.to_string(),
+            MechState {
+                remaining_aps: mech_state.remaining_aps - points_consumed,
+                ..mech_state
+            },
+        );
+        state
+    }
+
+    fn reset_mech_actions(state: &MatchState) -> MatchState {
```

Comment:
Again, slightly misleading not sure what "actions" means, but `action_points` is a little more clear.